### PR TITLE
WIP: V3PushAppAndConfirm integration helper and associated helpers

### PR DESCRIPTION
**Pull request by drnic**
This PR provides a scaffold generated helper `V3PushAppAndConfirm`, and its implementation methods to create a cf v3 app, apply its manifest, push, and start the application.

Requirements:

* [ ] This PR uses `cf` CLI's internal `util/manifest` library for loading + writing a cf `manifest.yml` (so that the generated app name is written into the manifest) - this requires a cli PR https://github.com/cloudfoundry/cli/pull/1653

This PR, plus #29, are being dev/tested in https://github.com/drnic/sample3-sidecar-buildpack

### CODE REVIEW

**Comment by drnic** [comment]
I've updated this PR to avoid `cf v3-push --no-start`. With v3 commands we can attach env vars and start logging after `cf v3-create-app`; so we don't need to use `cf v3-push --no-start`.

**Comment by drnic** [comment]
I'm still blocked on the CLI team being confused about the nature of https://github.com/cloudfoundry/cli/pull/1653

Could you please talk to them about its requirements?

**Comment by Zanadar** [comment]
@drnic Thanks for this. Seems like a good change. I'm not sure how quickly we're going to be able to get to chatting with the CLI team, just fair warning, given our current prioritization around CNB work. Let us know if you make any headway on their side. If we talk to them I'll update you.

**Comment by shanks3012** [comment]
@abbychau

File: cutlass/cf.go [hunk 1]
```diff
@@ -356,23 +358,81 @@ func (a *App) PushNoStart() error {
...
 }
 
-func (a *App) V3Push() error {
-	if err := a.PushNoStart(); err != nil {
+func (a *App) V3CreateApp() error {
+	args := []string{"v3-create-app", a.Name}
+	command := exec.Command("cf", args...)
+	command.Stdout = DefaultStdoutStderr
+	command.Stderr = DefaultStdoutStderr
+	if err := command.Run(); err != nil {
```
Comment (dfreilich) [inline cutlass/cf.go]: Either return with a fmt.Errorf to clarify what the error is, or just return command.Run() to simplify the command.

File: cutlass/cf.go [hunk 2]
```diff
@@ -356,23 +358,81 @@ func (a *App) PushNoStart() error {
...
+	apps, err := manifest.ReadAndInterpolateManifest(a.Manifest, nil, nil)
+	if err != nil {
+		return err
+	}
+
+	// modify appname to generated test name
+	app := apps[0]
+	app.Name = a.Name
+
+	tmpManifestFile, err := ioutil.TempFile(os.TempDir(), "manifest-*.yml")
```
Comment (dfreilich) [inline cutlass/cf.go]: Use RandString (https://github.com/cloudfoundry/libbuildpack/blob/f7fe6466a8efe94915a0933a340ff087972ff8cf/util.go#L239-L246) to make it a more unique name, without a wildcard

File: cutlass/cf.go [hunk 3]
```diff
@@ -356,23 +358,81 @@ func (a *App) PushNoStart() error {
...
+	err = manifest.WriteApplicationManifest(app, tmpManifestFile.Name())
+	if err != nil {
+		return err
+	}
+
+	args := []string{"v3-apply-manifest", "-f", tmpManifestFile.Name()}
+	command := exec.Command("cf", args...)
+	command.Stdout = DefaultStdoutStderr
+	command.Stderr = DefaultStdoutStderr
+	if err := command.Run(); err != nil {
```
Comment (dfreilich) [inline cutlass/cf.go]: Either return with a fmt.Errorf to clarify what the error is, or just return command.Run() to simplify the command.

File: cutlass/cf.go [hunk 4]
```diff
@@ -356,23 +358,81 @@ func (a *App) PushNoStart() error {
...
+	}
+
+	args := []string{"v3-push", a.Name, "-p", a.Path}
 	command := exec.Command("cf", args...)
 	command.Stdout = DefaultStdoutStderr
 	command.Stderr = DefaultStdoutStderr
 	if err := command.Run(); err != nil {
 		return err
 	}
+
```
Comment (dfreilich) [inline cutlass/cf.go]: Either return with a fmt.Errorf to clarify what the error is, or just return command.Run() to simplify the command.

**Comment by abbychau** [comment]
@shanks3012 did you mention a wrong account?

**Comment by drnic** [comment]
I'll close this PR, since we now have `cf7` rather than `cf6 v3-xyz` commands. No doubt someone will come back to this topic another time. Sorry @dfreilich that I didn't get back to your suggestions. My bad.

**Pull request closed without merging**